### ISSUE TITLE
Tensorflow crashes when importing certain modules before tensorflow

### ISSUE THREAD

**Issue by jeherr** [tensorflow/tensorflow#13963]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 16.04

- **TensorFlow installed from (source or binary)**:
binary both from conda-forge and from pip install using the tfBinaryURL

- **TensorFlow version (use command below)**:
1.3.0

- **Python version**:
3.6.2 from Anaconda

- **CUDA/cuDNN version**:
CUDA 8.0 cuDNN 6.0

- **GPU model and memory**:
GTX 1060 6 GB RAM

- **Exact command to reproduce**:
```python
import psi4
import tensorflow as tf
hello = tf.constant('Hello, TensorFlow!')
sess = tf.Session()
print(sess.run(hello))
```

### Describe the problem
Running the above code produces the following error

```
2017-10-24 22:17:35.570620: F tensorflow/core/framework/op.cc:165] Non-OK-status: RegisterAlreadyLocked(deferred_[i]) status: Invalid argument: Could not parse default value '4000' from Attr("upper_frequency_limit: float = 4000") for Op Mfcc
Could not parse default value '20' from Attr("lower_frequency_limit: float = 20") for Op Mfcc
forrtl: error (76): Abort trap signal
```

Strangely if tensorflow is imported before psi4 then the same error does not occur. Also, the error seems to be related to one of the Ops in tensorflow/tensorflow/core/kernels/mfcc.h which as far as I can tell is not an Op that I'm even using.

This issue occurs with at least two different python modules that I know of, psi4 as I have run into, and lycon which has their own issue open regarding this same error ethereon/lycon#3. In the linked issue it seems that this error occurs for others on python 2 and 3.5 with Ubuntu 16.04 both with and without GPU support for tensorflow, but another user states that this did not occur for him on Ubuntu 14.04 or on macOS 10.13. It seems like this may be an issue specific to at least Ubuntu 16.04 then.

### Source code / logs
Full output from the error

```
2017-10-24 22:17:35.570620: F tensorflow/core/framework/op.cc:165] Non-OK-status: RegisterAlreadyLocked(deferred_[i]) status: Invalid argument: Could not parse default value '4000' from Attr("upper_frequency_limit: float = 4000") for Op Mfcc
Could not parse default value '20' from Attr("lower_frequency_limit: float = 20") for Op Mfcc
forrtl: error (76): Abort trap signal
Image              PC                Routine            Line        Source             
libpcm.so.1        00007FCA4B14A725  Unknown               Unknown  Unknown
libpcm.so.1        00007FCA4B148347  Unknown               Unknown  Unknown
libpcm.so.1        00007FCA4B05FAA2  Unknown               Unknown  Unknown
libpcm.so.1        00007FCA4B05F8F6  Unknown               Unknown  Unknown
libpcm.so.1        00007FCA4B02DEFD  Unknown               Unknown  Unknown
libpcm.so.1        00007FCA4B031ABF  Unknown               Unknown  Unknown
libpthread.so.0    00007FCA5609F390  Unknown               Unknown  Unknown
libc.so.6          00007FCA55CF9428  Unknown               Unknown  Unknown
libc.so.6          00007FCA55CFB02A  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0DA71B94  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0D94AAFE  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0D94B21A  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0D924199  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0D946EFE  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0D934760  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0D68F8D7  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0D62136E  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0D621488  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0D621876  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0B615970  Unknown               Unknown  Unknown
_pywrap_tensorflo  00007FCA0B434177  Unknown               Unknown  Unknown
python             000055BE1E6850C6  Unknown               Unknown  Unknown
python             000055BE1E7181DC  Unknown               Unknown  Unknown
python             000055BE1E7398BA  Unknown               Unknown  Unknown
python             000055BE1E711FEB  Unknown               Unknown  Unknown
python             000055BE1E7182B5  Unknown               Unknown  Unknown
python             000055BE1E7398BA  Unknown               Unknown  Unknown
python             000055BE1E713116  Unknown               Unknown  Unknown
python             000055BE1E713B84  Unknown               Unknown  Unknown
python             000055BE1E684F8E  Unknown               Unknown  Unknown
python             000055BE1E73AF74  Unknown               Unknown  Unknown
python             000055BE1E711334  Unknown               Unknown  Unknown
python             000055BE1E712221  Unknown               Unknown  Unknown
python             000055BE1E7182B5  Unknown               Unknown  Unknown
python             000055BE1E7398BA  Unknown               Unknown  Unknown
python             000055BE1E711908  Unknown               Unknown  Unknown
python             000055BE1E712221  Unknown               Unknown  Unknown
python             000055BE1E7182B5  Unknown               Unknown  Unknown
python             000055BE1E7398BA  Unknown               Unknown  Unknown
python             000055BE1E711FEB  Unknown               Unknown  Unknown
python             000055BE1E7182B5  Unknown               Unknown  Unknown
python             000055BE1E7398BA  Unknown               Unknown  Unknown
python             000055BE1E711334  Unknown               Unknown  Unknown
python             000055BE1E712221  Unknown               Unknown  Unknown
python             000055BE1E7182B5  Unknown               Unknown  Unknown
python             000055BE1E7398BA  Unknown               Unknown  Unknown
python             000055BE1E712D58  Unknown               Unknown  Unknown
python             000055BE1E713B0C  Unknown               Unknown  Unknown
python             000055BE1E78DF04  Unknown               Unknown  Unknown
python             000055BE1E65475A  Unknown               Unknown  Unknown
python             000055BE1E6548FD  Unknown               Unknown  Unknown
python             000055BE1E65495E  Unknown               Unknown  Unknown
python             000055BE1E656B65  Unknown               Unknown  Unknown
python             000055BE1E65951E  Unknown               Unknown  Unknown
libc.so.6          00007FCA55CE4830  Unknown               Unknown  Unknown
python             000055BE1E7403D9  Unknown               Unknown  Unknown
Aborted (core dumped)
```

**Comment by allenlavoie** [comment]
(I'm assuming this is a symbol conflict based on the symptoms you've posted; I haven't looked into it thoroughly, so feel free to correct me)

The good news is that newer versions of TensorFlow (1.4+) will not cause problems like this, since we're no longer loading our symbols into the global symbol table. The bad news is that other libraries which export symbols via RTLD_GLOBAL can still cause problems for TensorFlow.

You could theoretically work around this problem by adding RTLD_DEEPBIND to the dlopen call in pywrap_tensorflow.py. This would break workflows which depend on redefining TensorFlow symbols (LD_PRELOAD), so we don't do it by default.

So basically this is contributions welcome if you have a reasonable way for us to avoid this symbol conflict (i.e. not contorting too much). Otherwise you can file a bug with the other library (since they're the ones exporting symbols globally; but do make sure you can reproduce in TF 1.4, since TF 1.3 can definitely cause these problems).

**Comment by yaroslavvb** [comment]
@jeherr some relevant background on fixing this kind of crash due to PyTorch. Similar solution would be to encourage psi4 to stop exporting unnecessary symbols into global symbol namespace

**Comment by jeherr** [comment]
I'll try to test this with the current Tensorflow 1.4 branch once I get some time. Should be able to soon.

**Comment by tatatodd** [comment]
@jeherr did you get around to trying this on TensorFlow 1.4?

**Comment by jeherr** [comment]
Sorry I meant to get back to this sooner. So I've been working on getting tensorflow 1.4 installed. I was using python 3.6 and it's seems that I ran into issue #14182. So until that gets worked out I decided I would try to use python 3.5. So I set up a new conda environment, installed everything I needed, and now I'm hitting a "terminate called after throwing an instance of 'std::bad_cast'" error, which I am pretty sure is stemming from psi4 at this point. 

Since this is just a personal project for me, I won't have the time to try to work out this error I'm getting with psi4 and python 3.5. So at this point I think I am stuck waiting until a fix comes for the tensorflow 1.4 issue with python 3.6 is resolved.

**Comment by allenlavoie** [comment]
For what it's worth, the "Python versions don't match" warning from Cython is harmless. You can try it on 3.6 with TF 1.4.

**Comment by tatatodd** [comment]
Marking as "community support" for now, in accordance with issue triage.

**Comment by allenlavoie** [comment]
What's the error? The QT/matplotlib one? The warning gets printed unconditionally, so it would be printed before any other failure even if that other failure isn't related.

**Comment by easyfly007** [comment]
I also meet this issue, tensorflow /keras (with tensorflow backend) abort if I import some module before. (some internal module).

so I move import tensorflow and keras to the very begining then the issue disappeared.

**Comment by allenlavoie** [comment]
This isn't a TensorFlow issue. Import order can work around other libraries exporting conflicting symbols, or RTLD_DEEPBIND can let TF ignore them. Closing this to keep the tracker focused on actionable issues.
